### PR TITLE
r/aws_evidently_project

### PR DESCRIPTION
**Pull request by GlennChia**
### Community Note

* Please vote on this pull request by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original pull request comment to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for pull request followers and do not help prioritize the request

Relates #22624

Output from acceptance testing:

```
$ make testacc TESTARGS='-run=TestAccEvidentlyProject' PKG=evidently
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./internal/service/evidently/... -v -count 1 -parallel 20  -run=TestAccEvidentlyProject -timeout 180m
=== RUN   TestAccEvidentlyProject_basic
=== PAUSE TestAccEvidentlyProject_basic
=== RUN   TestAccEvidentlyProject_tags
=== PAUSE TestAccEvidentlyProject_tags
=== RUN   TestAccEvidentlyProject_updateDataDeliveryCloudWatchLogGroup
=== PAUSE TestAccEvidentlyProject_updateDataDeliveryCloudWatchLogGroup
=== RUN   TestAccEvidentlyProject_updateDataDeliveryS3Bucket
=== PAUSE TestAccEvidentlyProject_updateDataDeliveryS3Bucket
=== RUN   TestAccEvidentlyProject_updateDataDeliveryS3Prefix
=== PAUSE TestAccEvidentlyProject_updateDataDeliveryS3Prefix
=== RUN   TestAccEvidentlyProject_updateDataDeliveryCloudWatchToS3
=== PAUSE TestAccEvidentlyProject_updateDataDeliveryCloudWatchToS3
=== RUN   TestAccEvidentlyProject_disappears
=== PAUSE TestAccEvidentlyProject_disappears
=== CONT  TestAccEvidentlyProject_basic
=== CONT  TestAccEvidentlyProject_updateDataDeliveryS3Prefix
=== CONT  TestAccEvidentlyProject_disappears
=== CONT  TestAccEvidentlyProject_tags
=== CONT  TestAccEvidentlyProject_updateDataDeliveryCloudWatchLogGroup
=== CONT  TestAccEvidentlyProject_updateDataDeliveryS3Bucket
=== CONT  TestAccEvidentlyProject_updateDataDeliveryCloudWatchToS3
--- PASS: TestAccEvidentlyProject_disappears (18.96s)
--- PASS: TestAccEvidentlyProject_basic (40.58s)
--- PASS: TestAccEvidentlyProject_updateDataDeliveryCloudWatchLogGroup (49.93s)
--- PASS: TestAccEvidentlyProject_updateDataDeliveryCloudWatchToS3 (53.82s)
--- PASS: TestAccEvidentlyProject_updateDataDeliveryS3Prefix (55.10s)
--- PASS: TestAccEvidentlyProject_updateDataDeliveryS3Bucket (55.52s)
--- PASS: TestAccEvidentlyProject_tags (59.43s)
PASS
ok      github.com/hashicorp/terraform-provider-aws/internal/service/evidently  59.499s
```

### CODE REVIEW

File: internal/service/cloudwatchevidently/project_test.go [hunk 1]
```diff
@@ -0,0 +1,419 @@
...
+	for _, rs := range s.RootModule().Resources {
+		if rs.Type != "aws_cloudwatchevidently_project" {
+			continue
+		}
+
+		input := &cloudwatchevidently.GetProjectInput{
+			Project: aws.String(rs.Primary.ID),
+		}
+
+		resp, err := conn.GetProject(input)
```
Comment (DrFaust92) [inline internal/service/cloudwatchevidently/project_test.go:279]: extract to finder and use it instead in all places where GetProject is used

File: internal/service/cloudwatchevidently/project_test.go [hunk 2]
```diff
@@ -0,0 +1,419 @@
...
+		Providers:    acctest.Providers,
+		CheckDestroy: testAccCheckProjectDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccProjectConfig_basic(rName, originalDescription),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckProjectExists(resourceName, &project),
+					resource.TestCheckResourceAttrSet(resourceName, "active_experiment_count"),
+					resource.TestCheckResourceAttrSet(resourceName, "active_launch_count"),
+					resource.TestCheckResourceAttrSet(resourceName, "arn"),
```
Comment (DrFaust92) [inline internal/service/cloudwatchevidently/project_test.go:40]: use acctest.CheckResourceAttrRegionalARN

File: internal/service/cloudwatchevidently/project_test.go [hunk 3]
```diff
@@ -0,0 +1,419 @@
...
+					resource.TestCheckResourceAttrSet(resourceName, "status"),
+					resource.TestCheckResourceAttr(resourceName, "tags.%", "1"),
+					resource.TestCheckResourceAttr(resourceName, "tags.Key1", "Test Project"),
+				),
+			},
+		},
+	})
+}
+
+func TestAccProject_updateTags(t *testing.T) {
```
Comment (DrFaust92) [inline internal/service/cloudwatchevidently/project_test.go:80]: follow https://github.com/hashicorp/terraform-provider-aws/blob/main/docs/contributing/contribution-checklists.md#resource-tagging-acceptance-testing-implementation conevntion

**Review by DrFaust92 — CHANGES_REQUESTED** [review]

**Comment by DrFaust92** [comment]
See linting errors, package and resource name should be `evidently` without the cloudwatch prefix

**Comment by GlennChia** [comment]
Thanks @DrFaust92 for the review, refactored the functions, naming, and tests accordingly. Acceptance test results:

```
$ make testacc TESTARGS='-run=TestAccEvidentlyProject' PKG=evidently
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./internal/service/evidently/... -v -count 1 -parallel 20  -run=TestAccEvidentlyProject -timeout 180m
=== RUN   TestAccEvidentlyProject_basic
=== PAUSE TestAccEvidentlyProject_basic
=== RUN   TestAccEvidentlyProject_tags
=== PAUSE TestAccEvidentlyProject_tags
=== RUN   TestAccEvidentlyProject_updateDataDelivery
=== PAUSE TestAccEvidentlyProject_updateDataDelivery
=== RUN   TestAccEvidentlyProject_disappears
=== PAUSE TestAccEvidentlyProject_disappears
=== CONT  TestAccEvidentlyProject_basic
=== CONT  TestAccEvidentlyProject_updateDataDelivery
=== CONT  TestAccEvidentlyProject_tags
=== CONT  TestAccEvidentlyProject_disappears
--- PASS: TestAccEvidentlyProject_disappears (17.41s)
--- PASS: TestAccEvidentlyProject_updateDataDelivery (27.17s)
--- PASS: TestAccEvidentlyProject_basic (37.38s)
--- PASS: TestAccEvidentlyProject_tags (54.36s)
PASS
ok      github.com/hashicorp/terraform-provider-aws/internal/service/evidently  54.422s
```

**Review by DrFaust92 — APPROVED** [review]
LGTM

**Comment by GlennChia** [comment]
Most recent test results after changes

```bash
$ make testacc TESTARGS='-run=TestAccEvidentlyProject' PKG=evidently
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./internal/service/evidently/... -v -count 1 -parallel 20  -run=TestAccEvidentlyProject -timeout 180m
=== RUN   TestAccEvidentlyProject_basic
=== PAUSE TestAccEvidentlyProject_basic
=== RUN   TestAccEvidentlyProject_tags
=== PAUSE TestAccEvidentlyProject_tags
=== RUN   TestAccEvidentlyProject_updateDataDeliveryCloudWatchLogGroup
=== PAUSE TestAccEvidentlyProject_updateDataDeliveryCloudWatchLogGroup
=== RUN   TestAccEvidentlyProject_updateDataDeliveryS3Bucket
=== PAUSE TestAccEvidentlyProject_updateDataDeliveryS3Bucket
=== RUN   TestAccEvidentlyProject_updateDataDeliveryS3Prefix
=== PAUSE TestAccEvidentlyProject_updateDataDeliveryS3Prefix
=== RUN   TestAccEvidentlyProject_updateDataDeliveryCloudWatchToS3
=== PAUSE TestAccEvidentlyProject_updateDataDeliveryCloudWatchToS3
=== RUN   TestAccEvidentlyProject_disappears
=== PAUSE TestAccEvidentlyProject_disappears
=== CONT  TestAccEvidentlyProject_basic
=== CONT  TestAccEvidentlyProject_updateDataDeliveryS3Prefix
=== CONT  TestAccEvidentlyProject_disappears
=== CONT  TestAccEvidentlyProject_tags
=== CONT  TestAccEvidentlyProject_updateDataDeliveryCloudWatchLogGroup
=== CONT  TestAccEvidentlyProject_updateDataDeliveryS3Bucket
=== CONT  TestAccEvidentlyProject_updateDataDeliveryCloudWatchToS3
--- PASS: TestAccEvidentlyProject_disappears (18.96s)
--- PASS: TestAccEvidentlyProject_basic (40.58s)
--- PASS: TestAccEvidentlyProject_updateDataDeliveryCloudWatchLogGroup (49.93s)
--- PASS: TestAccEvidentlyProject_updateDataDeliveryCloudWatchToS3 (53.82s)
--- PASS: TestAccEvidentlyProject_updateDataDeliveryS3Prefix (55.10s)
--- PASS: TestAccEvidentlyProject_updateDataDeliveryS3Bucket (55.52s)
--- PASS: TestAccEvidentlyProject_tags (59.43s)
PASS
ok      github.com/hashicorp/terraform-provider-aws/internal/service/evidently  59.499s
```

**Review by ewbankkit — APPROVED** [review]
LGTM :rocket:.

```console
% make testacc TESTARGS='-run=TestAccEvidentlyProject_' PKG=evidently ACCTEST_PARALLELISM=2
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./internal/service/evidently/... -v -count 1 -parallel 2  -run=TestAccEvidentlyProject_ -timeout 180m
=== RUN   TestAccEvidentlyProject_basic
=== PAUSE TestAccEvidentlyProject_basic
=== RUN   TestAccEvidentlyProject_tags
=== PAUSE TestAccEvidentlyProject_tags
=== RUN   TestAccEvidentlyProject_updateDataDeliveryCloudWatchLogGroup
=== PAUSE TestAccEvidentlyProject_updateDataDeliveryCloudWatchLogGroup
=== RUN   TestAccEvidentlyProject_updateDataDeliveryS3Bucket
=== PAUSE TestAccEvidentlyProject_updateDataDeliveryS3Bucket
=== RUN   TestAccEvidentlyProject_updateDataDeliveryS3Prefix
=== PAUSE TestAccEvidentlyProject_updateDataDeliveryS3Prefix
=== RUN   TestAccEvidentlyProject_updateDataDeliveryCloudWatchToS3
=== PAUSE TestAccEvidentlyProject_updateDataDeliveryCloudWatchToS3
=== RUN   TestAccEvidentlyProject_disappears
=== PAUSE TestAccEvidentlyProject_disappears
=== CONT  TestAccEvidentlyProject_basic
=== CONT  TestAccEvidentlyProject_updateDataDeliveryS3Prefix
--- PASS: TestAccEvidentlyProject_basic (40.47s)
=== CONT  TestAccEvidentlyProject_updateDataDeliveryCloudWatchLogGroup
--- PASS: TestAccEvidentlyProject_updateDataDeliveryS3Prefix (52.68s)
=== CONT  TestAccEvidentlyProject_updateDataDeliveryS3Bucket
--- PASS: TestAccEvidentlyProject_updateDataDeliveryCloudWatchLogGroup (47.96s)
=== CONT  TestAccEvidentlyProject_tags
--- PASS: TestAccEvidentlyProject_updateDataDeliveryS3Bucket (51.97s)
=== CONT  TestAccEvidentlyProject_disappears
--- PASS: TestAccEvidentlyProject_disappears (16.88s)
=== CONT  TestAccEvidentlyProject_updateDataDeliveryCloudWatchToS3
--- PASS: TestAccEvidentlyProject_tags (58.53s)
--- PASS: TestAccEvidentlyProject_updateDataDeliveryCloudWatchToS3 (48.69s)
PASS
ok  	github.com/hashicorp/terraform-provider-aws/internal/service/evidently	176.417s
```

**Comment by ewbankkit** [comment]
@GlennChia Thanks for the contribution :tada: :clap:.

**Pull request merged**